ed-odyssey-materials-helper: 2.156 -> 2.173

elfenermarcell opened the pull request:
https://github.com/jixxed/ed-odyssey-materials-helper/releases/tag/2.165

The auto update failed due to a patch failing to apply, which I've updated.
Although I'm not exactly sure the auto updates are going to work though - I don't think it updates `deps.json`?
## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ToasterUwU commented:
I will review it later, but for the deps.json file, maybe the nix-update setting on the package can do it.
Its a "how to update this" for the update bot, so assuming its versatile enough, you could add this and the auto update PRs would work.

ToasterUwU commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 402566`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ed-odyssey-materials-helper</li>
  </ul>
</details>

elfenermarcell commented:
I think I've managed to fix the updatescript: nix-update only updates the version and hash, but I added a step after to generate a new lockfile too.

ToasterUwU commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 402566`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ed-odyssey-materials-helper</li>
  </ul>
</details>

ToasterUwU commented:
Eval failed, but that seems to be a CI issue, not a issue with the PR itself.

elfenermarcell commented:
I have figured out why it doesn't build on aarch64. The `javafx` dependencies have a seperate aarch64 version. Therefore a seperate lockfile would be needed for aarch64. A passthru.updateScript that can create both lockfiles on a single machine would also be needed otherwise updating would be annoying. However I haven't found a way to do that so I marked the package as only supporting `x86_64-linux` (for now)

FliegendeWurst commented on pkgs/by-name/ed/ed-odyssey-materials-helper/package.nix:
```diff
@@ -115,6 +125,6 @@ stdenv.mkDerivation rec {
     ];
     mainProgram = "ed-odyssey-materials-helper";
-    platforms = lib.platforms.linux;
+    platforms = [ "x86_64-linux" ];
```
Have you checked whether an update script like https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/ke/keyboard-layout-editor/package.nix#L37-L44 would work?

elfenermarcell replied:
That does in fact work, thanks!

elfenermarcell committed:
ed-odyssey-materials-helper: 2.156 -> 2.173 and fix updateScript

Explicitly marked the package as supporting x86_64-linux and aarch64-linux,
since the lockfile will be built for these platforms and I can test on them.

ToasterUwU commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 402566`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>ed-odyssey-materials-helper</li>
  </ul>
</details>
